### ISSUE TITLE
zoom to extend

### ISSUE THREAD

**Issue by jukos** [andreaordonselli/qgis2o.gis#21]
Hello,
the function “zoom to extend” is not displayed.
How can I display it?
How can I display it on mobile?
Best regards

Version 2.23

**Comment by andreaordonselli** [comment]
Do you mean "zoom to extent"?
![image](https://github.com/user-attachments/assets/d4ea71a3-6e5e-40cb-a1a7-b4c308e7d0de)

In desktop view it should be visible in each vector layer by default.
In the smartphone view, since there is little space in the layerswitcher, I disabled it, do you want to re-enable it?

**Comment by jukos** [comment]
> Do you mean "zoom to extent"?

Yes


> In desktop view it should be visible in each vector layer by default.

Unfortunately not.


> In the smartphone view, since there is little space in the layerswitcher, I disabled it, do you want to re-enable it?

Yes, I would like to see this to test it for myself.

**Comment by andreaordonselli** [comment]
You should send me your qgis project to understand why the "zoom to extent" button does not appear.
To reactivate the elements in the smartphone view follow this issue
https://github.com/andreaordonselli/qgis2o.gis/issues/15

To view "zoom to extent" you must write:
```css
.ol-touch .layerExtent {
		display: inline-table;
		right: ..... px;
	}
```

**Comment by jukos** [comment]
It doesn't matter what I do.
Different QGIS versions, different projects, new test project with one layer, tests on various browsers. No display of zoom to extent.

**Comment by andreaordonselli** [comment]
Many thanks for your report. The bug is expected to be fixed in the next release.
For now you can find this code in the `o.gis.js` file and modify it

before:
```javascript
	map.once('rendercomplete', function(layer) {
		setLayersExtent(); 
	});
```

after:
```javascript
	map.once('rendercomplete', function(layer) {
		setLayersExtent(); 
		layerSwitcher.drawPanel(); 
	});
```

**Comment by andreaordonselli** [comment]
This issue has been resolved in the v2.24.0 release
Thanks for your report
